NSubstitute.DbConnection/NSubstitute.DbConnection.Tests/QueryParameterMatchingTests.cs

```diff
@@ -249,4 +250,86 @@ public void ShouldFailWhenSetUpWithNoParameters()
         });
         result.Should().Throw<NotSupportedException>().WithMessage(NoMatchingQueryErrorMessage+$": '{commandText}'");
     }
+
+    [Test]
+    public void ShouldIncludeOutputParametersWhenMatching()
+    {
+        var mockConnection = Substitute.For<IDbConnection>().SetupCommands();
+        var commandText = "spMySproc";
+        var record = new KeyValueRecord(1, "abc");
+        mockConnection.SetupQuery(commandText)
+            .WithParameters(new Dictionary<string, object> { { "id", 1 } })
+            .WithOutputParameters(("output", 7))
+            .Returns(record);
+
+        using var command = mockConnection.CreateCommand();
+        command.CommandText = commandText;
+        command.AddParameter("id", 1);
+        command.AddOutputParameter("output", DbType.Byte);
+        mockConnection.Open();
+
+        command.ExecuteNonQuery().Should().Be(1);
+        command.Parameters["output"].As<DbParameter>().Value.Should().Be(7);
+    }
+
+    [Test]
+    public void ShouldIncludeOutputParametersWhenMatchingQuery()
+    {
+        var mockConnection = Substitute.For<IDbConnection>().SetupCommands();
+        var commandText = "select * from table";
+
+        var record = new KeyValueRecord(1, "abc");
+        mockConnection.SetupQuery(commandText)
+            .WithOutputParameters(("output", "example"))
+            .Returns(record);
+        
+        using var command = mockConnection.CreateCommand();
+        command.CommandText = commandText;
+        command.AddOutputParameter("output", DbType.String);
```

Comment:
Answering my previous question, I guess not since the only thing you specify when adding an output parameter to the command is its type, never its value?

Reply:
Yes, the value passed in with the output parameters is the mock output value. I can rename it to be more explicit.
You _can_ have in/out parameters, those are distinct from output parameters. I'm not actually sure what the real code does if you pass a value to an out parameter...